### DISCUSSION TITLE
Need your help to scrap a particular website rejecting my requests

### DISCUSSION THREAD

**Discussion by Fluidetom** [FreshRSS/FreshRSS#6876] [Q&A] [answered]
Hey guys,

I'm having trouble to scrape a particular website listing job offers.
The devs have recently updated their website and my feed no longer works.
I'm now trying to update the feed but I keep getting an error in FreshRSS telling me the feed encountered an error.
I also tried to delete the existing one and create a brand new subscription, same result.

- The page I'm trying to scape : https://environnement.brussels/offres-emploi
- Finding new items :  `//li[@class="list-cards__cards-card"]`
- Title : `descendant::h3`
- Content : `"See website"`
- URL : `descendant::div[@class="card-category__bottom"]/a/@href`

If anyone could have a look, I really don't know what I'm missing here.

Is it possible that the website rejects the FreshRSS requests ?

Anyway, thanks in advance for your time.
Much appreciated

**Comment by Frenzie** [comment]
`list-cards__cards-card` doesn't occur on that page without Javascript.

At a glance I think the data's actually in the following element:
```html
<div data-baseline-vue-component-id="vacancy-overview" id="vue-a99613bc-432e-4e8a-919d-8dfc1be04b42" class="js-vue-component"><vacancy-overview :filters="{&quot;degrees&quot;:[{&quot;label&quot;:&quot;Enseignement secondaire sup\u00e9rieur&quot;,&quot;id&quot;:&quot;1530&quot;},{&quot;label&quot;:&quot;Pas de dipl\u00f4me&quot;,&quot;id&quot;:&quot;1531&quot;}],&quot;job_types&quot;:[{&quot;label&quot;:&quot;Poste sp\u00e9cifique&quot;,&quot;id&quot;:&quot;1559&quot;}]}" :items="[{&quot;link&quot;:&quot;https:\/\/environnement.brussels\/jobs\/ouvrieres-forestieres-ouvriers-forestiers-hfx-convention-de-premier-emploi-25-ans&quot;,&quot;degrees&quot;:[&quot;Pas de dipl\u00f4me&quot;],&quot;job_types&quot;:[&quot;Poste sp\u00e9cifique&quo etc.
```

Since it seems to be a Drupal website maybe they could enable RSS if you ask them? A few possible candidate URLs like `/rss` and `/rss.xml` aren't presently available.

**Comment by Fluidetom** [comment]
Hi,

Tx for your reply.
I'm not a dev so I'm  just trying to guess what you're saying ;-)

You mean that the code I see in the console is generated by Javascript ?
Is that an issue for FreshRSS XPath scraping?

When looking at the code, I selected `<li class="list-cards__cards-card">` as this is where each job offer is listed.
This is how I proceed for other websites, and it is usually working like a charm.

Each of these <li ...> are actually under the node you mentioned `<div data-baseline-vue-component-id="vacancy-overview" id="vue-a99613bc-432e-4e8a-919d-8dfc1be04b42" class="js-vue-component" data-v-app=""> `

![Screenshot from 2024-10-10 12-04-50](https://github.com/user-attachments/assets/c5d38c85-a8d9-4061-8a8f-62fe54757c8d)

I've also emailed the webmaster in case of they'd accept to enable RSS feed, but I'd like to understand what's blocking the XPath scraping here.

Thanks in advance

**Reply by Alkarex** [reply]
To rephrase what @Frenzie was saying: you cannot use your Web browser DOM inspector as you do, because this shows a page after JavaScript transformation.

To accurately observe the response, use a *View source* instead, or download the HTML page and open it in a text editor.

The relevant job informations are in a JSON structure, which you cannot scrape with just HTML+XPath, because it is not HTML.

For now, the easiest would we to use a bridge for instance with https://github.com/RSS-Bridge/rss-bridge

We might add a possibility to extract JSON from HTML pages in the future, but this is not something FreshRSS can at the moment (besides with an extension).

**Reply by Alkarex** [reply]
Please open a new feature request for a mode *HTML+JSON* to support your usecase

**Reply by Alkarex** [reply, marked as answer]
I have a patch coming:

![image](https://github.com/user-attachments/assets/731a4cc1-b67d-4ab6-96c0-66f971702782)

Edit: See https://github.com/FreshRSS/FreshRSS/pull/6888

**Reply by Alkarex** [reply]
See https://github.com/FreshRSS/FreshRSS/pull/6888

**Comment by Fluidetom** [comment]
Hi guys, 
Sorry i've been AFK the last couple of days.
Thanks so much for your prompt action and for the patch, this is great !
I did had a look at RSS-bridge in the meantime, but frankly, for a non-dev person this is a bit overwhelming.
Anyway, I'll keep an eye for the patch and install it as soon as it's available.
Is there any special instruction on how to use this new HTML+Json feed type ?

Again, thanks a million to you all, This is really amazing ! Muche appreciated

**Reply by Alkarex** [reply]
It is already available in your rolling release *edge* branch, so you are welcome to start using it and give feedback

**Comment by Fluidetom** [comment]
Hi @Alkarex and happy new year to all,

So, I've updated my self-hosted FreshRSS instance to v1.25.0 and I now have the option to add an _HTML+XPATH+JSON_ feed.

So 'I'm trying to list all job offers posted here : https://environnement.brussels/offres-emploi.
But I'm not a dev and I never worked with JSON before.

I know all posts are listed under `<ul class="list-cards__cards">` but I don't know how I must reference the JSON etc ...
I had a look at the source of the page, but I've to be honest, this is beyond my basic HTML knowledge.

Could someone help me to configure a feed to fetch all job offers from that page please?

Thanks in advance

**Reply by Frenzie** [reply]
Save the example from https://github.com/FreshRSS/FreshRSS/pull/6888#issuecomment-2408643363 as `leefmilieubrussel.opml` and then you can go to `+` (Add feed) → `Import / Export`

**Reply by Fluidetom** [reply]
thanks a lot @Frenzie @Alkarex  I got it working 

![Screenshot from 2025-01-10 10-50-43](https://github.com/user-attachments/assets/04d5b0cd-f317-43be-9928-a116d72ff8d3)

One last comment from a non-dev user like me.
I discovered XPath when I started using FreshRSS. It looks easy and logical enough so that even a newbie could find the required info via the DOM explorer. This new type of feed with the JSON seems a bit more tricky to apprehend for non-dev users.

Do you think it would be possible to write a one-pager guide in the project documentation (or reference an existing one), to explain : 

1. how to identify a situation where a HTML+XPath+JSON+DotNotation feed type must be used?
2. the basic steps to configure such feed, and how to retrieve the required parameters?

Anyway, for the time being, thanks a million for the support, and the change deployed.
It is really much appreciated!

Have all a nice day !

**Reply by Frenzie** [reply]
1. Only as a last resort. ;-) Also try a few typical URLs like `/feed` (basically any WP site), `/rss` (typical Drupal), `/rss.xml`, `/atom`, etc.
2. You open the page, right click, developer tools, inspect (or Ctrl+Shift+i). Or just "view source" as a first step.
  Then go to the sources tab (in Chromia browsers) and press Ctrl+Shift+f, type `Stagiaire`.
  You then see where it actually occurs.
